taskana-adapter-camunda-system-connector/src/main/java/pro/taskana/adapter/systemconnector/camunda/api/impl/CamundaSystemConnectorImpl.java

```diff
@@ -84,13 +83,19 @@ public String retrieveVariables(String taskId) {
     }
 
     @Override
-    public List<ReferencedTask> retrieveFinishedTasks(Instant finishedAfter) {
-        return taskRetriever.retrieveFinishedCamundaTasks(camundaSystemURL.getSystemTaskEventUrl(), finishedAfter);
+    public List<ReferencedTask> retrieveTerminatedTasks() {
+        return taskRetriever.retrieveTerminatedCamundaTasks(camundaSystemURL.getSystemTaskEventUrl());
     }
 
     @Override
-    public void taskanaTasksHaveBeenCreatedForReferencedTasks(List<ReferencedTask> referencedTasks) {
-        taskEventCleaner.taskanaTasksHaveBeenCreatedForReferencedTasks(referencedTasks,
+    public void taskanaTasksHaveBeenCreatedForNewReferencedTasks(List<ReferencedTask> referencedTasks) {
+        taskEventCleaner.cleanEventsForReferencedTasks(referencedTasks,
+            camundaSystemURL.getSystemTaskEventUrl());
+    }
+
+    @Override
+    public void taskanaTasksHaveBeenTerminatedForTerminatedReferencedTasks(List<ReferencedTask> referencedTasks) {
```

Comment:
Does this really mean termindated? Or do we deal with completed task here? If they are the completed task I would prefer to be precise in the naming.